lit/docs/guides/pipeline-guides.lit

```diff
@@ -0,0 +1,676 @@
+    \codeblock{yaml}{{
+    jobs:
+    - name: the-job
+      plan: # Get some artifacts
+      - get: apples
+      - get: apple-basket
+      - get: monkeys
+      - put: apples-in-basket
+        input: [apples, apple-basket] # specify the exact inputs needed
+        params:
+          apples-location: apples/location
+          basket: apple-basket
+    }}
+
+  }
+
+  \section{
+    \title{Putting Task Config's in Files}
+
+    A lot of the pipeline examples that you will find on this site and in
+    resource repos will embed a \reference{schema.step.task-step.config} directly
+    in the pipeline. This is a nice way of clearly seeing what inputs/outputs the
+    task uses. Tasks are usually designed to be used in multiple places, maybe
+    with slightly different configuration. To support this scenario, most users
+    store task configs in files instead of embedding the config directly in the
+    pipeline.
+
+    Here's what this looks like in practice:
+
+    \titled-codeblock{todays-date.yml, stored in some git repo under a folder named "task"}{yaml}{{
+    platform: linux
+
+    image_resource: # define a default image for the task to use
+      type: registry-image
+      source:
+        repository: busybox
+
+    run:
+      path: date
+      args: ["+%Y-%m-%d"]
+    }}
+    \inset{} {- spacer -}
+
+    Using the task in a pipeline:
+
+    \titled-codeblock{pipline.yml}{yaml}{{
+    resources:
+    - name: ci
+      type: git
+      source:
+        uri: http://github.com/username/my-ci-repo
+
+    jobs:
+    - name: the-best-job
+      plan:
+      - get: ci
+      - task: what-day-is-it
+        file: ci/tasks/todays-date.yml
+    }}
+  }
+
+  \section{
+    \title{\code{Get} Images for Tasks Instead of using Anonymous Image Resources}
+
+    It is easy to let Concourse fetch images for tasks right when they are
+    needed, it's the easy out-of-the-box solution. Another way is to pass the
+    image(s) for tasks as inputs to jobs.
+
+    \bold{Before - Anonymous Image Fetching for Tasks}
+
+    \codeblock{yaml}{{
+    jobs:
+    - name: job
+      plan:
+      - task: simple-task
+        config:
+          platform: linux
+          image_resource: # anonymous image resource
+            type: registry-image
+            source:
+              repository: busybox
+          run:
+            path: echo
+            args: ["Hello world!"]
+    }}
+
+    \inset{} {- spacer -}
+
+    \bold{After - Passing Task Image as Job Inputs}
+
+    \codeblock{yaml}{{
+    resources:
+    - name: busybox
+      type: registry-image
+      source:
+        repository: busybox
+        username: ((docker.user))
+        password: ((docker.password))
+
+    jobs:
+    - name: job
+      plan:
+      - get: busybox # pass image into job
+      - task: simple-task
+        image: busybox # use image for task. Overrides anonymous image
+        config:
+          platform: linux
+          image_resource: # overridden by step-level "image" field
```

Comment:
I feel like this would be less confusing to newcomers if we left the `image_resource` out in this example. We're introducing it as a one-or-the-other thing, and the fact that it takes precedence is maybe better left to just the docs.

Speaking of, maybe we can link to the relevant places in the docs for `image_resource` and `task.image` here too?

Reply:
Really great suggestion. Changed the first paragraph to link to both.